### PR TITLE
Unit Testing

### PR DESCRIPTION
**Pull request by Have-Samuel**
In this PR We:
- Wrote unit tests using Rspec.

### CODE REVIEW

File: spec/person_rspec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  it 'should have a can_use_services?' do
+    expect(@person.can_use_services?).to eq true
+  end
+
+  it 'should have a correct_name' do
+    expect(@person.correct_name).to eq 'SomeOne'
+  end
+end
```
Comment (AnselemOdims) [inline spec/person_rspec.rb:23]: - [ ] You are required to add tests for all methods. Please write tests for the `add_rental` method on this class.

File: spec/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  it 'should have author' do
+    expect(@book.author).to eq 'Jon Doe'
+  end
+
+  it 'should have rentals' do
+    expect(@book.rentals).to eq []
+  end
+end
```
Comment (AnselemOdims) [inline spec/book_spec.rb:19]: - [ ] Please kindly write tests for the `add_rental` and `to_json` methods on this class.

File: spec/decorator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+describe Decorator do
+  before(:each) do
+    @decorator = Decorator.new('Sediqullah')
+  end
+
+  it 'should have a nameable' do
+    expect(@decorator.nameable).to eq('Sediqullah')
+  end
+end
```
Comment (AnselemOdims) [inline spec/decorator_spec.rb:12]: - [ ] As pointed out earlier, please let's write test for the `correct_name` method on this class.

File: spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  it 'should have a can_use_services?' do
+    expect(@student.can_use_services?).to eq true
+  end
+
+  it 'should have a correct_name' do
+    expect(@student.correct_name).to eq 'Someone'
+  end
+end
```
Comment (AnselemOdims) [inline spec/student_spec.rb:32]: - [ ] Please kindly include tests for the method `play_hooky` and the setter method `change_classroom`.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @team, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/book_spec.rb (hunk 2 above)
Comment (sediqullahbadakhsh) [reply spec/book_spec.rb:19]: Thanks, Tests added

File: spec/decorator_spec.rb (hunk 3 above)
Comment (sediqullahbadakhsh) [reply spec/decorator_spec.rb:12]: thanks testes added

File: spec/student_spec.rb (hunk 4 above)
Comment (sediqullahbadakhsh) [reply spec/student_spec.rb:32]: Thanks, tests added

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢🟢🟢 

Hi @Have-Samuel & @sediqullahbadakhsh ,

Good job working on the changes highlighted by the previous reviewer. 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights

✅ Correct use of Github workflow
✅ No linter error
✅ Tests for all classes and methods passed 👍  

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**